src/main/java/frc/robot/Robot.java

```diff
@@ -24,13 +27,23 @@ public void robotPeriodic() {
   }
 
   @Override
-  public void disabledInit() {}
+  public void disabledInit() {
+    m_neutralModeTimer.start();
+  }
 
   @Override
-  public void disabledPeriodic() {}
+  public void disabledPeriodic() {
+    // if DISABLED_COAST_DELAY seconds have passed since disabling, set the neutral mode to coast
+    if (m_neutralModeTimer.get() > RobotConstants.DISABLED_COAST_DELAY) {
+      m_robotContainer.setDisabledNeutralMode();
+    }
+  }
 
   @Override
-  public void disabledExit() {}
+  public void disabledExit() {
+    m_robotContainer.setEnabledNeutralMode();
+    m_neutralModeTimer.reset();
```

Comment:
If you reset without also calling `stop`, the timer will keep running while the robot is enabled, so will instantly be triggered when the robot is disabled again.

Reply:
Alternatively, the reset can go in disabledInit.